keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -515,25 +518,44 @@ Similar to the BucketAccess for COSI created bucket, this BucketAccess should re
+
+Once a delete has been issued to a bucket, no new BucketAccesses can be created for it. Buckets having valid BucketAccesses (Buckets in use) will not be deleted until all the BucketAccesses are cleaned up.
 
 Buckets can be created with one of two deletion policies:
  - Retain
  - Delete
 
 When the deletion policy is Retain, then the underlying bucket is not cleaned up when the Bucket object is deleted. When the deletion policy is Delete, then the underlying bucket is cleaned up when the Bucket object is deleted.
 
-Only when all accessors (BucketAccesses) of the Bucket are deleted, is the Bucket itself cleaned up. There is a finalizer on the Bucket that prevents it from being deleted until all the accessors are done using it. 
+Only when all accessors (BucketAccesses) of the Bucket are deleted, is the Bucket itself cleaned up. There is a finalizer on the Bucket that prevents it from being deleted until all the accessors are done using it.
+
+When a user deletes a BucketAccess, the corresponding secret/serviceaccount are also deleted. If a pod has that secret mounted when delete is called, then a finalizer on the secret will prevent it from being deleted. Instead, the deletionTimestamp will be set on the secret. In this way, access to a Bucket is preserved until the application pod dies.
+
+When an admin deletes any of the class objects, it does not affect existing Buckets as fields from the class objects are copied into the Buckets during creation.
+
+If a Bucket is manually deleted by an admin, then a finalizer on the Bucket prevents it from being deleted until the binding BucketClaim is deleted.
+
+## Storage Quotas
+
+Quotas provide administrators with the ability to limit the total storage space used.
+
+### Bucket Quotas
+
+Rook users have requested the ability to limit the max size of buckets as the max number of objects in a bucket.
+However, adding this to the COSI API spec may be problematic.
 
-When a user deletes a BucketAccess, the corresponding secret/serviceaccount are also deleted. If a pod has that secret mounted when delete is called, then a finalizer on the secret will prevent it from being deleted. Instead, the deletionTimestamp will be set on the secret. In this way, access to a Bucket is preserved until the application pod dies. 
+Amazon's S3 service does not seem to implement either of those quotas. Either by their
+[quota mechanism](https://docs.aws.amazon.com/general/latest/gr/s3.html) or by their
+[bucket policy mechanism](https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-bucket-policies.html).
 
-When an admin deletes any of the class objects, it does not affect existing Buckets as fields from the class objects are copied into the Buckets during creation. 
+These particular quotas may not be a good fit for the COSI design. :(
```

Comment:
This is unfortunate, but I wonder if it would still be beneficial to continue with some comments about quotas, even if the guidance is that the opaque parameters should be used on a per-provider basis.

Alternatively the COSI design could include these as official API and make the implementation of them optional for backends. To do that, we would probably want to follow some sort of mechanism for querying/reporting supported functionality, similar to this in CSI: https://kubernetes-csi.github.io/docs/support-fsgroup.html#supported-modes